### DISCUSSION TITLE
Typescript debug support for API Routes

### DISCUSSION THREAD

**Discussion by conradkoh** [vercel/next.js#16893] [Help]
# Feature request

## Is your feature request related to a problem? Please describe.

When creating a typescript file for an api route, for example at path `pages/api/users.ts`, and starting next using `NODE_OPTIONS='--inspect' next dev`, an applied breakpoint brings you to a generate typescript file instead of the file you are debugging.

## Describe the solution you'd like

The debugger should bring me to the place where I placed the breakpoint.

## Describe alternatives you've considered

Considered writing a manual file called server.js to start api routes in debug mode when trying to debug an error instead, but I would need to remap the API calls to a separate port and change the configuration in the frontend files.

## Additional context

-

**Comment by schalk-b** [comment]
It looks like your question has gotten lost. But this is already supported, all you need to do is add:
```
"compilerOptions": {
    ...
    "sourceMap": true
  },
```

And follow the rest of the documentation for debugging:
https://nextjs.org/docs/advanced-features/debugging
